content/docs/rfcs/14/README.md

```diff
@@ -111,6 +117,20 @@ Whisper/Waku v1 envelopes are compatible with Waku v2 messages format.
 Waku v2 implements a pub/sub messaging pattern over libp2p.
 This makes redundant some Whisper/Waku v1 envelope fields (e.g., `expiry`, `ttl`, `topic`, etc.), so they can be ignored.
 
+# Deterministic message hashing
+
+In Protocol Buffers v3, the deterministic serialization is not canonical across the different implementations and languages.
+It is also unstable across different builds with schema changes due to unknown fields. 
+
+To overcome this interoperability limitation, a Waku v2 message's hash MUST be computed following this schema:
+
+```
+message_hash = sha256(concat(pubsub_topic, message.payload, message.content_topic, message.meta))
```

Comment:
a test vector for this would be good to ensure all implementations are aligned. Cc @LNSD